### DISCUSSION TITLE
Check if a PDF page has only text or other additions too

### DISCUSSION THREAD

**Discussion by vignesh0710** [pymupdf/PyMuPDF#3757] [Looking for help] [answered]
In a given pdf page, is it possible to check if the ```page``` contains 

**CASE A:**                              ```only text``` or 
**ALL THE OTHER CASES**: it ```contains text in a different format``` or ```text with other additions```

**Examples for text ```with other format``` or ```other additions``` can include**

1.  Pages with graphs/images
2. Pages with table
3. Pages with multiple text boxes
4. Text in columns format

**Following Approaches I Tried did not work:**

1. Checking for annotations
2. Check if the page has rows/columns
3. check if the page has tables

DISCLAIMER: All the screenshots are from publicly available document
[https://www2.deloitte.com/content/dam/Deloitte/nl/Documents/financial-services/deloitte-nl-fsi-fintech-report-1.pdf](url)

**Example for page with ```only text```**

<img width="616" alt="Screenshot 2024-08-07 at 10 00 50 PM" src="https://github.com/user-attachments/assets/33fb2ee8-522e-4cb8-852c-9243e1fabb77">

**Examples for text ```with other format``` or ```other additions``` can include**

1.  Pages with graphs/images
2. Pages with table
3. Pages with multiple text boxes
4. Text in columns format

<img width="972" alt="Screenshot 2024-08-07 at 10 04 33 PM" src="https://github.com/user-attachments/assets/97faafa7-08f3-4b7b-b647-80b50fbde77d">

<img width="1189" alt="Screenshot 2024-08-07 at 10 04 45 PM" src="https://github.com/user-attachments/assets/5a8fdccd-f84f-4553-99a4-f50b344b47b2">

<img width="789" alt="Screenshot 2024-08-07 at 10 05 03 PM" src="https://github.com/user-attachments/assets/bd68af17-eb5c-47e9-a127-87ba1437efb4">

<img width="865" alt="Screenshot 2024-08-07 at 10 05 12 PM" src="https://github.com/user-attachments/assets/d90cd41f-f48c-489d-89c9-baebba7ad0a6">

<img width="841" alt="Screenshot 2024-08-07 at 10 05 26 PM" src="https://github.com/user-attachments/assets/94e2ed65-4d1c-4407-93f8-0385fae0f348">

<img width="517" alt="Screenshot 2024-08-07 at 10 05 46 PM" src="https://github.com/user-attachments/assets/0087d8bd-e332-4fca-aec3-a32577db2923">

Any suggestions will be helpful, thanks

**Comment by JorjMcKie** [comment, marked as answer]
There are a number of checks available:
* page.get_text() in various variations. The "text" variant offers a quick check for any text at all or only white text.
* page.first_annot, page.first_link, page.first_widget: if any of them is not `None`, then there are objects of the respective type.
* page.get_drawings != [] means that there exist drawings / vector graphics
* page.get_image_info() != [] there are images on page
* page.find_tables().tables != [] there are tables on the page

Text columns on the page can not be directly determined.
